pkgs/development/compilers/ponyc/default.nix

```
@@ -3,22 +3,22 @@
 
 stdenv.mkDerivation (rec {
   pname = "ponyc";
-  version = "0.50.0";
+  version = "0.51.2";
 
   src = fetchFromGitHub {
     owner = "ponylang";
     repo = pname;
     rev = version;
-    sha256 = "sha256-FnzlFTiJrqoUfnys+q9is6OH9yit5ExDiRszQ679QbY=";
+    sha256 = "sha256-hQvdVXNGEEbr4a0wjp/kEpdDsTXaRCpq+8jXUvTEZEI=";
 
     fetchSubmodules = true;
   };
 
   ponygbenchmark = fetchFromGitHub {
     owner = "google";
     repo = "benchmark";
-    rev = "v1.5.4";
-    sha256 = "1dbjdjzkpbsq3jl9ksyg8mw759vkac8qzq1557m73ldnavbhz48x";
+    rev = "v1.5.6";
```

```suggestion
    rev = "v1.5.6";
```
google-benchmark is at [v1.7.0](https://github.com/google/benchmark/releases/tag/v1.7.0).
Is there a particular reason for this version?
Is there a reason google-benchmark package is not part of nixpkgs?

---

The information you seek is in lib/CMakeLists.txt

https://github.com/ponylang/ponyc/blob/0e144cd6dd621a1434a21dc6d0dbeda2e5e35342/lib/CMakeLists.txt#L33-L36

The values are hard-coded into the build-system.

---

> You can use the hardcoded version, but then add a comment to point to source of package version. (As you just told me.)
Also, as the package has inner dependencies, it will require an update script to be upgrade-able by bot. So, it is desirable to test a general package because of this. I understand you have made an attempt, but if it is possible to package package in isolation, it makes the upgrade process easier (no need for special upgrade script).